Imath/ImathBox.h

```diff
@@ -317,62 +352,106 @@ Box<T>::majorAxis() const
 
 template <typename T> class Box;
 
+///
+/// The Box<Vec2<T>> template represents a 2-dimenional bounding box defined by
+/// minimum and maximum values of type Vec2<T>. The min and max members are
+/// public.
+///
+
 template <class T> class Box<Vec2<T>>
 {
   public:
     //-------------------------
     //  Data Members are public
     //-------------------------
 
+    /// The minimum value of the box
     Vec2<T> min;
+
+    /// The maximum value of the box
     Vec2<T> max;
 
     //-----------------------------------------------------
     //  Constructors - an "empty" box is created by default
     //-----------------------------------------------------
 
+    /// Construct an empty bounding box.
     IMATH_HOSTDEVICE IMATH_CONSTEXPR14 Box();
+
+    /// Construct a bounding box that contains a single point.
     IMATH_HOSTDEVICE IMATH_CONSTEXPR14 Box (const Vec2<T>& point);
+
+    /// Construct a bounding box with the given minimum and maximum values.
     IMATH_HOSTDEVICE IMATH_CONSTEXPR14 Box (const Vec2<T>& minT, const Vec2<T>& maxT);
 
     //--------------------
     //  Operators:  ==, !=
     //--------------------
 
+    /// Compare two Box objects for equality
     IMATH_HOSTDEVICE constexpr bool operator== (const Box<Vec2<T>>& src) const;
+
+    /// Compare two Box objects for inequality
     IMATH_HOSTDEVICE constexpr bool operator!= (const Box<Vec2<T>>& src) const;
 
     //------------------
     //  Box manipulation
     //------------------
 
+    /// Set the Box to be empty. A Box is empty if the mimimum is greater
+    /// than the maximum. makeEmpty() sets mimimum to T::baseTypeMax() and
+    /// maximum to T::baseTypeMin().
     IMATH_HOSTDEVICE void makeEmpty();
+
+    /// Extend the Box to include the given point.
     IMATH_HOSTDEVICE void extendBy (const Vec2<T>& point);
+
+    /// Extend the Box to include the given box.
     IMATH_HOSTDEVICE void extendBy (const Box<Vec2<T>>& box);
+
+    /// Make the box include the entire range of T
```

Comment:
Probably should decide one way or the other whether comments that look like sentences should end with a period. Some do here, some don't.

Reply:
Since some comments are multiple sentences, ending all sentences with a . seems most consistent.